### PR TITLE
stm32: clock_control: Add test suite for stm32h7

### PR DESCRIPTION
**Pull request by erwango**
Similarly to what was done for U5 in https://github.com/zephyrproject-rtos/zephyr/pull/40731,
add some tests dedicated to H7 clock_control driver configuration

### CODE REVIEW

**Comment by erwango** [comment]
This PR requires https://github.com/zephyrproject-rtos/zephyr/pull/41304 to pass the tests

File: tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@
 cmake_minimum_required(VERSION 3.20.0)
 
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
-project(stm32_clock_configuration)
+project(stm32_clock_configuration_h5)
```
Comment (FRASTM) [inline tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt:6]: Is the `project(stm32_clock_configuration_h5) `or project(stm32_clock_configuration_**u**5)

**Review by FRASTM — COMMENTED** [review]
LGTM

File: tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt (hunk 1 above)
Comment (erwango) [reply tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt:6]: Right, this is now fixed. Txs

**Review by FRASTM — APPROVED** [review]

File: tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt (hunk 1 above)
Comment (ABOSTM) [reply tests/drivers/clock_control/stm32_clock_configuration/stm32u5/CMakeLists.txt:6]: Still need to be changed to stm32_clock_configuration_u5

File: tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_24.overlay [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Warning: This overlay performs configuration from clean sheet.
+ * It is assumed that is is applied after clear_clocks.overlay file.
```
Comment (ABOSTM) [inline tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_24.overlay:9]: typo: "it is"

File: tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_48.overlay [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Warning: This overlay performs configuration from clean sheet.
+ * It is assumed that is is applied after clear_clocks.overlay file.
```
Comment (ABOSTM) [inline tests/drivers/clock_control/stm32_clock_configuration/stm32u5/boards/msis_48.overlay:9]: dito

File: dts/arm/st/h7/stm32h7.dtsi [hunk 4]
```diff
@@ -46,9 +46,6 @@
 		clk_hse: clk-hse {
 			#clock-cells = <0>;
 			compatible = "st,stm32-hse-clock";
-			/* H7 clock driver may not always need this value */
```
Comment (ABOSTM) [inline dts/arm/st/h7/stm32h7.dtsi:49]: As per former comment, I understand that the property is required by binding, even if board doesn't have HSE at all.
By removing this default value, it becomes necessary for board without HSE, to declare this property.

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: dts/arm/st/h7/stm32h7.dtsi (hunk 4 above)
Comment (erwango) [reply dts/arm/st/h7/stm32h7.dtsi:49]: This property is required by binding only if node is enabled (it is disabled by default).

**Comment by erwango** [comment]
@ABOSTM, thanks for the review, I think all is fixed now

**Review by ABOSTM — APPROVED** [review]

**Pull request merged**